compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1614,6 +1615,26 @@ fn linker_with_args<'a, B: ArchiveBuilder<'a>>(
     let cmd = &mut *codegen_results.linker_info.to_linker(base_cmd, &sess, flavor);
     let link_output_kind = link_output_kind(sess, crate_type);
 
+    if let LinkerFlavor::Lld(LldFlavor::Gcc) = flavor {
+        #[cfg(target_os = "linux")]
+        {
+            let tools_path =
+                sess.host_filesearch(PathKind::All).get_tools_search_paths(crt_objects_fallback);
+            // FIXME handle rust-lld not found
+            let lld_path = tools_path.into_iter().find(|p| p.join("rust-lld").exists()).unwrap();
+            // FIXME handle error instead of silently ignoring
+            let _ = std::os::unix::fs::symlink(lld_path.join("rust-lld"), lld_path.join("ld"));
```

Comment:
You can't just create a symlink in the sysroot here. It may be read-only if rustc is installed globally. Even if it isn't, you shouldn't silently modify the sysroot. Instead you should probably create this symlink in the build system during the dist step for rust-lld.

Reply:
@bjorn3 Sorry, I'm not sure what you mean by build system. Do you mean the directory my clone of the repo is in? Also, by dist step, do you mean in `src/bootstrap/dist.rs`?

Reply:
> I'm not sure what you mean by build system.

The build system is the code in `src/bootstrap`.

> Also, by dist step, do you mean in src/bootstrap/dist.rs?

Yes

Reply:
Wait, how would that work? Suppose I'm using a copy of rustc installed as a binary from my distro. Where should I create the symlink then?

Reply:
The symlink should be in `$SYSROOT/lib/rustlib/$TARGET/bin/` next to `rust-lld`.

Reply:
Huh, I remember using your suggestion to find the sysroot [here](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/How.20to.20get.20sysroot.20and.20HOST_TARGET_TRIPLE), but then I discovered that snippet produced identical results on my machine without having to refactor any code. When would the two differ?

Reply:
I don't think they would. I was just using a more explicit path in my comment. The point is that the ld symlink should be in the same dir as rust-lld.

Reply:
Wait, so what would I need to do then if your suggestion was the same as my original path?

Reply:
The suggestion is to create the symlink in src/bootstrap/dist.rs at build time instead of in link.rs at runtime.

Reply:
I see `dist.rs` prepares a tarball of rust for final packaging. But when I am testing, if I create the symlink in `dist.rs`, will the symlink be created for the test?

Reply:
No, it won't. My bad. It should be in `src/bootstrap/compile.rs` after the copy just after `if let Some(lld_install) = lld_install {` I think.